### ISSUE TITLE
not webserver agnostic, too much of rewrite dependences

### ISSUE THREAD

**Issue by mckaygerhard** [GR8DAN/markdown-CMS#2]
your CMS depends so much of the crap named apache and mod rewrite.. 

curently many webhostings are migrating to lighttpd, cherokee, hiawatta and nginx and your softare still thinks apache2 is a defacto standar .. 

there's the two problems:

1. all the routes are into "/" of the webserver, cannot handel root hiyacking or subpaths (this last most important)
2. all the routes need a complete rewrite and cannot be embebed into others integrations, neither be portable on webserver migrations/agnostic

**Comment by GR8DAN** [comment]
Thanks for raising this issue. When the project was started in 2015 the dominant webservers were Apache and IIS. These days Nginx has a significant share of the market. Therefore a _.conf_ file for Nginx is needed. Support for the many other types of webservers is useful, though most have less than 0.1% of market share, however, _.conf_ files for them would be welcomed.

Yes, as with many CMSs, the assumption is that webservers have been configured correctly to mitigate path attacks (a potential issue for inexperienced admins). Many CMSs could benefit from being less reliant on server configs.

Dependency on URL rewriting is not ideal and needs to be addressed for better encapsulation/portability. The embedded use case was not an original consideration, however, others have mentioned it before and it needs revisiting. The current workaround is via subdomains. Migrations haven't been a particular issue as _mv/xcopy_ is generally without issue, however, portability considerations need to be investigated.

**Comment by mckaygerhard** [comment]
> Thanks for raising this issue. When the project was started in 2015 the dominant webservers were Apache and IIS. These days Nginx has a significant share of the market. Therefore a _.conf_ file for Nginx is needed. Support for the many other types of webservers is useful, though most have less than 0.1% of market share, however, _.conf_ files for them would be welcomed.

wrong.. mostly production servers used lighttpd form static and nginx for dinamic ones.. 

also  when you made software for "marked share" your are not so free software and only supoprt the monopoly way oif the things

> 
> Yes, as with many CMSs, the assumption is that webservers have been configured correctly to mitigate path attacks

to much dependence of the webserver.. as a not good practice .. i am not so happy with this project now.. sorry

**Comment by GR8DAN** [comment]
> wrong.. mostly production servers used lighttpd form static and nginx for dinamic ones

The references I use for Apache and Nginx market share are from [here](https://news.netcraft.com/archives/2021/08/25/august-2021-web-server-survey.html) and [here](https://w3techs.com/technologies/overview/web_server). A link to alternative market share statistics is welcomed. However, as I said in the paragraph you quoted, _.conf_ files for any webserver are welcomed.

> when you made software for "marked share" your are not so free software and only supoprt the monopoly way oif the things

I find it strange when technical issues are perceived to be against a certain point of view. This software works with common web hosting packages, its original intention. This issue is for a technical enhancement to support Nginx, which now has a high market share, and it is increasing all the time and will become the new monopoly. Contributions to enable this software to support any webserver (e.g., Lighttpd) are welcomed. I.e. free software welcomes contributions from everyone, whatever their reasons for writing the software.

> to much dependence of the webserver.. as a not good practice

Which I acknowledge and am happy for any contribution:

> Dependency on URL rewriting is not ideal and needs to be addressed for better encapsulation/portability

Thank you for your comments. I hope you find a free CMS suitable for your needs amongst the many hundreds that are out there to choose from.
